xapian-applications/omega/date.cc

```diff
@@ -191,7 +192,13 @@ date_range_filter(const string & date_start, const string & date_end,
 {
     int y1, m1, d1, y2, m2, d2;
     if (!date_span.empty()) {
-	time_t secs = atoi(date_span.c_str()) * (24 * 60 * 60);
+	time_t secs;
+	unsigned int temp;
+	if (!parse_unsigned(date_span.c_str(), temp)) {
+	    throw "Datespan value must be >= 0";
+	} else {
+	    secs = temp * (24 * 60 * 60);
+	}
```

Comment:
You could lose the `else` (the `if` block will always transfer control away) and then you can move the declaration of `secs` down so it gets initialised at the point of declaration.  Being able to declare and initialise in one go is cleaner - it makes it completely obvious to the reader that the variable can't be used uninitialised.

Also, a more descriptive name than `temp` would make the code easier to understand - e.g. `days`.

Reply:
fixed in a fixup commit.